Change to gyp build process breaks install

I installed version `0.4.1` a little over a week ago, but unfortunately I cannot update or install `0.4.2` or later. I believe this is caused by changes to the gyp build instructions in 2ec40a1f12618c32812ec7dfe53bbb69573ae47a.

I noticed that you also switched from the reference implementation of argon2 to the optimized implementation in the process of changing the build. That's fine by me, and I can compile either version from [P-H-C/phc-winner-argon2](https://github.com/P-H-C/phc-winner-argon2/tree/4fe0d8cda37691228dd5a96a310be57369403a4b) without error when I do it myself. It's not entirely clear to me what gyp is doing differently.

In the future, please bump the minor version number when you make significant changes to the build system. IMO, patch releases should contain only extremely conservative changes. While I understand that you might look at it as a patch release because there are no API changes, as a library user what I really care about is that my code doesn't break in a patch update.

Here is the error I'm encountering:

```
vagrant@machine:~/ws/myapp$ npm update --save argon2
npm WARN install Couldn't install optional dependency: Unsupported

> argon2@0.4.2 preinstall /home/vagrant/ws/myapp/node_modules/.staging/argon2-8e1864def6601eebe8a65bded53fa406
> git submodule update --init


> argon2@0.4.2 postinstall /home/vagrant/ws/myapp/node_modules/argon2
> npm run build


> argon2@0.4.2 build /home/vagrant/ws/myapp/node_modules/argon2
> node-gyp configure rebuild

make: Entering directory '/home/vagrant/ws/myapp/node_modules/argon2/build'
  CC(target) Release/obj.target/libargon2/argon2/src/argon2.o
  CC(target) Release/obj.target/libargon2/argon2/src/core.o
../argon2/src/core.c: In function ‘validate_inputs’:
../argon2/src/core.c:364:35: warning: comparison of unsigned expression < 0 is always false [-Wtype-limits]
         if (ARGON2_MIN_PWD_LENGTH > context->pwdlen) {
                                   ^
../argon2/src/core.c:394:31: warning: comparison of unsigned expression < 0 is always false [-Wtype-limits]
         if (ARGON2_MIN_SECRET > context->secretlen) {
                               ^
../argon2/src/core.c:409:34: warning: comparison of unsigned expression < 0 is always false [-Wtype-limits]
         if (ARGON2_MIN_AD_LENGTH > context->adlen) {
                                  ^
  CC(target) Release/obj.target/libargon2/argon2/src/blake2/blake2b.o
  CC(target) Release/obj.target/libargon2/argon2/src/thread.o
  CC(target) Release/obj.target/libargon2/argon2/src/encoding.o
  CC(target) Release/obj.target/libargon2/argon2/src/opt.o
In file included from ../argon2/src/opt.c:23:0:
../argon2/src/blake2/blamka-round-opt.h: In function ‘fBlaMka’:
../argon2/src/blake2/blamka-round-opt.h:40:1: warning: SSE vector return without SSE enabled changes the ABI [-Wpsabi]
 static BLAKE2_INLINE __m128i fBlaMka(__m128i x, __m128i y) {
 ^
In file included from ../argon2/src/opt.h:18:0,
                 from ../argon2/src/opt.c:20:
/usr/lib/gcc/i686-linux-gnu/5/include/emmintrin.h:1029:1: error: inlining failed in call to always_inline ‘_mm_add_epi64’: target specific option mismatch
 _mm_add_epi64 (__m128i __A, __m128i __B)
 ^
In file included from ../argon2/src/opt.c:23:0:
../argon2/src/blake2/blamka-round-opt.h:42:12: error: called from here
     return _mm_add_epi64(_mm_add_epi64(x, y), _mm_add_epi64(z, z));
            ^
In file included from ../argon2/src/opt.h:18:0,
                 from ../argon2/src/opt.c:20:
/usr/lib/gcc/i686-linux-gnu/5/include/emmintrin.h:1029:1: error: inlining failed in call to always_inline ‘_mm_add_epi64’: target specific option mismatch
 _mm_add_epi64 (__m128i __A, __m128i __B)
 ^
In file included from ../argon2/src/opt.c:23:0:
../argon2/src/blake2/blamka-round-opt.h:42:12: error: called from here
     return _mm_add_epi64(_mm_add_epi64(x, y), _mm_add_epi64(z, z));
            ^
In file included from ../argon2/src/opt.h:18:0,
                 from ../argon2/src/opt.c:20:
/usr/lib/gcc/i686-linux-gnu/5/include/emmintrin.h:1029:1: error: inlining failed in call to always_inline ‘_mm_add_epi64’: target specific option mismatch
 _mm_add_epi64 (__m128i __A, __m128i __B)
 ^
In file included from ../argon2/src/opt.c:23:0:
../argon2/src/blake2/blamka-round-opt.h:42:12: error: called from here
     return _mm_add_epi64(_mm_add_epi64(x, y), _mm_add_epi64(z, z));
            ^
In file included from ../argon2/src/opt.h:18:0,
                 from ../argon2/src/opt.c:20:
/usr/lib/gcc/i686-linux-gnu/5/include/emmintrin.h:1131:1: error: inlining failed in call to always_inline ‘_mm_mul_epu32’: target specific option mismatch
 _mm_mul_epu32 (__m128i __A, __m128i __B)
 ^
In file included from ../argon2/src/opt.c:23:0:
../argon2/src/blake2/blamka-round-opt.h:41:19: error: called from here
     const __m128i z = _mm_mul_epu32(x, y);
                   ^
libargon2.target.mk:89: recipe for target 'Release/obj.target/libargon2/argon2/src/opt.o' failed
make: *** [Release/obj.target/libargon2/argon2/src/opt.o] Error 1
make: Leaving directory '/home/vagrant/ws/myapp/node_modules/argon2/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Linux 4.2.0-16-generic
gyp ERR! command "/usr/bin/nodejs" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "rebuild"
gyp ERR! cwd /home/vagrant/ws/myapp/node_modules/argon2
gyp ERR! node -v v5.4.1
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 

npm ERR! Linux 4.2.0-16-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "build"
npm ERR! node v5.4.1
npm ERR! npm  v3.3.12
npm ERR! code ELIFECYCLE
npm ERR! argon2@0.4.2 build: `node-gyp configure rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the argon2@0.4.2 build script 'node-gyp configure rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the argon2 package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp configure rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls argon2
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/vagrant/ws/myapp/node_modules/argon2/npm-debug.log
npm ERR! Linux 4.2.0-16-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "update" "--save" "argon2"
npm ERR! node v5.4.1
npm ERR! npm  v3.3.12
npm ERR! code ELIFECYCLE

npm ERR! argon2@0.4.2 postinstall: `npm run build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the argon2@0.4.2 postinstall script 'npm run build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the argon2 package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run build
npm ERR! You can get their info via:
npm ERR!     npm owner ls argon2
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/vagrant/ws/myapp/npm-debug.log
````

My system: Ubuntu 15.10, 32-bit

--- a/binding.gyp
+++ b/binding.gyp
@@ -1,4 +1,9 @@
 {
+  "target_defaults": {
+    "target_conditions": [
+      ["OS != 'win'", {"cflags": ["-msse","-msse2"]}]
+    ]
+  },
   "targets": [
     {
       "target_name": "libargon2",

